can_tools/scrapers/official/base.py

```diff
@@ -116,6 +116,116 @@ def _put_exec(self, engine: Engine, df: pd.DataFrame) -> None:
+    ) -> pd.DataFrame:
+        """Reshape columns in data to be long form definitions defined in `variable_map`.
+
+        Parameters
+        ----------
+        data :
+            Input data
+        variable_map : Union[str,int]
+            Map from column name to output variables
+
+        Returns
+        -------
+        data:
+            Reshaped DataFrame.
+        """
+        # parse out data columns
+        value_cols = list(set(data.columns) & set(variable_map.keys()))
+        assert len(value_cols) == len(variable_map)
+        id_vars = []
+        if "location_name" in data.columns:
+            id_vars.append("location_name")
+        if "location" in data.columns:
+            id_vars.append("location")
+        if "dt" in data.columns:
+            id_vars.append("dt")
+        if "vintage" in data.columns:
+            id_vars.append("vintage")
+
+        data = (
+            data.melt(id_vars=id_vars, value_vars=value_cols)
+            .dropna()
+            .assign(
+                value=lambda x: pd.to_numeric(
+                    x["value"].astype(str).str.replace(",", "")
+                ),
+            )
+            .pipe(self.extract_CMU, cmu=variable_map)
+            .drop(["variable"], axis=1)
+        )
+
+        if "vintage" not in data.columns:
+            data["vintage"] = self._retrieve_vintage()
+
+        return data
+
+    def _rename_or_add_date_and_location(
+        self,
+        data: pd.DataFrame,
+        location_name_column: Optional[str] = None,
+        location_column: Optional[str] = None,
+        location_names_to_drop: Optional[List[str]] = None,
+        date_column: Optional[str] = None,
+        date: Optional[pd.Timestamp] = None,
+        timezone: Optional[str] = None,
+    ):
+        """Renames or adds date and location columns.
+
+        Parameters
+        ----------
+        data :
+            Input data
+        location_name_column:
+            Name of column with location name
+        location_column:
+            Name of column with location (fips)
+        date_column:
+            Name of Column containing date.
+        date:
+            Date for data
+        timezone:
+            Timezone of data if date or date_column not supplied.
+
+        Returns
+        -------
+        data:
+            Data with date and location columns normalized.
+        """
+        assert location_name_column or location_column
+        assert date_column or date or timezone
+
+        rename_columns = {}
+        if location_name_column:
+            rename_columns[location_name_column] = "location_name"
+        if location_column:
+            rename_columns[location_column] = "location"
+        if date_column:
+            rename_columns[date_column] = "dt"
+
+        data = data.rename(columns=rename_columns)
+
+        if date_column:
+            data.loc[:, "dt"] = data["dt"].apply(lambda x: pd.to_datetime(x))
```

Comment:
Minor, but I think that having the rhs be `pd.to_datetime(data["dt"])` is same result, but potentially faster as it lets pandas vectorize